components/eam/src/physics/cam/gw_common.F90

```diff
@@ -742,4 +746,1472 @@ subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
+    enddo
+  enddo
+
+  do k = kts,kte
+    do i = its,ite
+      za(i,k) = 0.5_r8*(zq(i,k)+zq(i,k+1))
+    enddo
+  enddo
+
+  do i=its,ite
+    hpbl2 = hpbl(i)+10._r8
+    kpbl2 = kpbl(i)
+    kvar = 1
+    do k=kts+1,MAX(kpbl(i),kts+1)
+      IF (za(i,k)>300._r8) then
+        kpbl2 = k
+        IF (k == kpbl(i)) then
+          hpbl2 = hpbl(i)+10._r8
+        ELSE
+          hpbl2 = za(i,k)+10._r8
+        ENDIF
+        exit
+      ENDIF
+    enddo
+
+    if(xland1(i).gt.0._r8 .and. 2._r8*var(i).le.hpbl(i))then
+      if(br1(i).gt.0._r8 .and. thvx(i,kpbl2)-thvx(i,kts) > 0._r8)then
+        cleff    = sqrt(dxy(i)**2_r8 + dxyp(i)**2_r8)
+        cleff    = (2.0_r8/sncleff) * max(dxmax_ss,cleff)
+        coefm(i) = (1._r8 + ol(i)) ** (oa1(i)+1._r8)
+        xlinv(i) = coefm(i) / cleff
+        govrth(i)=g/(0.5_r8*(thvx(i,kpbl2)+thvx(i,kts)))
+        XNBV=sqrt(govrth(i)*(thvx(i,kpbl2)-thvx(i,kts))/hpbl2)
+
+        if(abs(XNBV/u1(i,kpbl2)).gt.xlinv(i))then
+          tauwavex0=0.5_r8*XNBV*xlinv(i)*(2._r8*MIN(var(i),varmax))**2_r8*ro(i,kvar)*u1(i,kvar)
+          tauwavex0=tauwavex0*ss_taper   ! "Scale-awareness"
+        else
+          tauwavex0=0._r8
+        endif
+
+        if(abs(XNBV/v1(i,kpbl2)).gt.xlinv(i))then
+          tauwavey0=0.5_r8*XNBV*xlinv(i)*(2._r8*MIN(var(i),varmax))**2._r8*ro(i,kvar)*v1(i,kvar)
+          tauwavey0=tauwavey0*ss_taper   ! "Scale-awareness"
+        else
+          tauwavey0=0._r8
+        endif
+
+        do k=kts,kpbl(i) !MIN(kpbl2+1,kte-1)
+          utendwave(i,k)=-1._r8*tauwavex0*2._r8*max((1._r8-za(i,k)/hpbl2),0._r8)/hpbl2
+          vtendwave(i,k)=-1._r8*tauwavey0*2._r8*max((1._r8-za(i,k)/hpbl2),0._r8)/hpbl2
+        enddo
+      endif
+    endif
+  enddo ! end i loop
+
+  do k = kts,kte
+    do i = its,ite
+      dudt(i,k)  = dudt(i,k) + utendwave(i,k)
+      dvdt(i,k)  = dvdt(i,k) + vtendwave(i,k)
+      dtaux2d_ss(i,k) = utendwave(i,k)
+      dtauy2d_ss(i,k) = vtendwave(i,k)
+      dusfc_ss(i) = dusfc_ss(i) + utendwave(i,k) * del(i,k)
+      dvsfc_ss(i) = dvsfc_ss(i) + vtendwave(i,k) * del(i,k)
+    enddo
+  enddo
+
+  ENDIF  ! end if gsd_gwd_ss == .true.
+  !================================================================
+  !add Beljaars et al. (2004, QJRMS, equ. 16) form drag:
+  !================================================================
+  IF (gsd_gwd_fd.and.(ss_taper.GT.1.E-02) ) THEN
+
+    utendform=0._r8
+    vtendform=0._r8
+    zq=0._r8
+
+    IF (.not.gsd_gwd_ss.and.(ss_taper.GT.1.E-02) ) THEN
+      ! Defining layer height. This is already done above is small-scale GWD is used
+      do k = kts,kte
+        do i = its,ite
+          zq(i,k+1) = dz2(i,k)+zq(i,k)
+        enddo
+      enddo
+
+      do k = kts,kte
+        do i = its,ite
+          za(i,k) = 0.5_r8*(zq(i,k)+zq(i,k+1))
+        enddo
+      enddo
+    ENDIF
+
+    DO i=its,ite
+      IF (xland1(i) .gt. 0..and.2._r8*var(i).gt.0) then
+        a1=0.00026615161_r8*var(i)**2_r8
+        a2=a1*0.005363_r8
```

Comment:
I see you're using a few "magic numbers" in this section. Do these come from a paper or something else we could cite here?